backend/wmg/data/snapshot.py

```diff
@@ -259,20 +303,31 @@ def _should_reload_snapshot(
     *,
     snapshot_schema_version: str,
     explicit_snapshot_id_to_load: Optional[str] = None,
-    read_versioned_snapshot: bool,
+    snapshot_local_disk_path: Optional[str] = None,
 ) -> tuple[bool, str]:
     """
     Returns a pair: (<should_reload>, <snapshot_id>) where <should_reload> is a boolean indicating
     whether then in-memory snapshot should be reloaded and <snapshot_id> is the id of the snapshot that
     the in-memory data structure represents.
     """
     snapshot_id = explicit_snapshot_id_to_load or _get_latest_snapshot_id(
-        snapshot_schema_version=snapshot_schema_version, read_versioned_snapshot=read_versioned_snapshot
+        snapshot_schema_version, snapshot_local_disk_path
     )
 
     if cached_snapshot is None:
         logger.info(f"Loading snapshot id: {snapshot_id}")
         return (True, snapshot_id)
+    # IMPORTANT NOTE: Updating cached_snapshot when snapshot_identifer value of
```

Comment:
- Under what conditions do we expect the deployment to fallback onto S3?
 - If we change S3's latest_snapshot_identifier, does the backend recognize that if we are still reading from local disk? Or does the backend only detect a change in S3's latest_snapshot_identifier if we are in fallback mode?

To rollback the snapshot when the deployment is in local disk mode, do we need to shell into the instance and edit the latest_snapshot_identifier file on disk?

Reply:
@atarashansky - 

- The only time the app will fallback to S3 is if `_local_disk_snapshot_is_valid()` returns `False`
- If `_local_disk_snapshot_is_valid()` returns `True`, the backend _will always read from local disk_. However, changes to `latest_snapshot_identifier` on S3 could affect what snapshot_id the app reads from local disk, if circumstances are such that the `snapshots` directory is downloaded onto the local disk _after `latest_snapshot_identifier` is changed on S3_.

This might happen if we manually redeploy after changing `latest_snapshot_identifier` on S3 - That is expected.

But a re-download of the `snapshots` directory would also happen if Fargate preempts a task and reschedules it on a new node. If prior to this reschedule, we update the `latest_snapshot_identifer` on S3, then this is reflected on the local disk. This has to do with the concern about data consistency and the [comments in this ticket](https://github.com/chanzuckerberg/single-cell-data-portal/issues/6403#issuecomment-1879432598) talk about it.

- To rollback when app is in local disk mode, nothing needs to change. Since we download the _entire `snapshots` directory_, we simply rollback by changing the schema version and/or snapshot_id in [wmg_api_config.py](https://github.com/chanzuckerberg/single-cell-data-portal/blob/main/backend/wmg/api/wmg_api_config.py)